e2e/test/MessageReceiveE2ETests.cs

```diff
@@ -234,6 +295,53 @@ public static async Task VerifyReceivedC2DMessageAsync(Client.TransportType tran
             Assert.IsTrue(received, $"No message received for device {deviceId} with payload={payload} in {FaultInjection.RecoveryTimeMilliseconds}.");
         }
 
+        public static async Task VerifyReceivedC2DMessageWithCancellationTokenAsync(Client.TransportType transport, DeviceClient dc, string deviceId, string payload, string p1Value)
+        {
+            Stopwatch sw = new Stopwatch();
+            bool received = false;
+
+            sw.Start();
+
+            Client.Message receivedMessage;
+
+            while (!received && sw.ElapsedMilliseconds < FaultInjection.RecoveryTimeMilliseconds)
+            {
+                _log.WriteLine($"Receiving messages for device {deviceId}.");
+
+                
+                receivedMessage = await dc.ReceiveAsync(new CancellationTokenSource(TIMESPAN_ONE_MINUTE).Token).ConfigureAwait(false);
```

Comment:
Why does this need to be 1 minute?
Can it be smaller value? As this is adding more time for tests. 

Adding any comments for the same will be helpful.

Reply:
No it won't add more time. If C2D function is good, it should be fast, otherwise it will get a `null` and fail the test in a minute. There is no SLA for C2D latency so one minute is the least interval we should wait.